### PR TITLE
DM-33936: Create GHAs to build containers and run unit tests.

### PR DESCRIPTION
**Pull request by ktlim**

### CODE REVIEW

**Review by kfindeisen — APPROVED** [review]
Looks good. Sorry for the delay; it took me a while to understand some bits.

File: .github/workflows/build-service.yml [hunk 1]
```diff
@@ -27,32 +27,32 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - name: Build image
-        run: docker build python/activator --tag $IMAGE_NAME --label "runnumber=${GITHUB_RUN_ID}"
-      - name: Login to GitHub Container Registry
+      - name: Login to Google Artifact Registry
         uses: docker/login-action@v1
```
[the remark this answers is not in the document]
Comment (ktlim) [reply .github/workflows/build-service.yml:55]: We need to authenticate to be able to pull the container for testing.

On the other hand, it's probably better to pull it from GitHub anyway.  I'll change it.

File: .github/workflows/build-service.yml [hunk 2]
```diff
@@ -1,32 +1,56 @@
...
+      - main
     paths:
       - '.github/workflows/build-service.yml'
       - 'python/activator/**'
   pull_request:
     paths:
       - '.github/workflows/build-service.yml'
       - 'python/activator/**'
-    branches:
-      - main
```
[the remark this answers is not in the document]
Comment (ktlim) [reply .github/workflows/build-service.yml]: It's on any update to the PR, including pushes to the branch that is being PRed.  Pushes to non-PR branches are more likely to be in-development and not actually ready for testing.

File: .github/workflows/build-base.yml [hunk 3]
```diff
@@ -5,12 +5,12 @@ on:
...
       - '.github/workflows/build-base.yml'
       - 'python/base/**'
+    branches:
+      - main
   pull_request:
     paths:
       - '.github/workflows/build-base.yml'
       - 'python/base/**'
-    branches:
-      - main
```
[the remark this answers is not in the document]
Comment (ktlim) [reply .github/workflows/build-base.yml]: Yeah, I don't know why I didn't squash that.

File: .github/workflows/build-service.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ktlim) [reply .github/workflows/build-service.yml:55]: Also added a comment in the workflow.

**Comment by ktlim** [comment]
Urgh.  Except that we've never pushed a "latest" to GitHub.  Let me see if I can fix that...

File: .github/workflows/build-service.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kfindeisen) [reply .github/workflows/build-service.yml]: I don't follow. Isn't "in development" exactly when you want to test, while a PR means it's ready for use?

[the remark this answers is not in the document]
Comment (ktlim) [reply .github/workflows/build-service.yml]: I typically use non-PR (or draft PR) branches as places to save random commits that may not actually function, creating the PR only when I expect it to actually work.

But the workflows as written allow you to manually trigger a build on any branch (or ref) you'd like: Actions tab, "Select workflow" button, "Run workflow" button, enter branch/tag/ref.

**Pull request merged**